src/Runner.Common/JobServerQueue.cs

```diff
@@ -292,14 +306,46 @@ private async Task ProcessWebConsoleLinesQueueAsync(bool runOnce = false)
                         {
                             try
                             {
-                                // we will not requeue failed batch, since the web console lines are time sensitive.
-                                if (batch[0].LineNumber.HasValue)
+                                bool pushedLinesViaWebsocket = false;
+                                if (this._websocketClient?.State == WebSocketState.Open)
                                 {
-                                    await _jobServer.AppendTimelineRecordFeedAsync(_scopeIdentifier, _hubName, _planId, _jobTimelineId, _jobTimelineRecordId, stepRecordId, batch.Select(logLine => logLine.Line).ToList(), batch[0].LineNumber.Value, default(CancellationToken));
+                                    var jsonData = JsonConvert.SerializeObject(new
```

Comment:
@TingluoHuang : `StringUtil.ConvertToJson` use this https://cs.github.com/actions/runner?q=StringUtil.ConvertToJson instead